---
title: "Summaries"
output: html_document
---

Now that we've cleaned up our taxonomy, let's take a look at the read distribution across the samples and remove samples with low read depths. 

Let's load our latest Phyloseq object 
```{r}
combinedPS_clean<-readRDS("Processed_data/combinedPS_clean.rds")
```

and needed packages 
```{r, message=FALSE,echo=TRUE}
library("microbiome") 
library("microbiomeutilities")
library("cowplot")
library("ggpubr")
```

As mentioned previously, there are plenty of packages available to analyse metabaroding data. To summarize the data the [Microbiome](https://microbiome.github.io/tutorials/cleaning_taxonomy_table.html) package provide a simple 




```{r}
microbiome::summarize_phyloseq(combinedPS_clean)
```


[microbiomutilities](https://microsud.github.io/microbiomeutilities/articles/microbiomeutilities.html)


```{r, message=FALSE}
Dep_raw<-plot_read_distribution(combinedPS_clean, 
                                groups = "seq_tech", 
                                plot.type = "histogram")+
  theme_biome_utils()+
  scale_x_continuous(trans='log10',
                   limits=c(1, 50000000))+
  scale_fill_manual(values=c("#111111"))+
  geom_vline(xintercept = 1000, 
             colour = "black", 
             linetype="dashed")+
  theme(legend.position="none")+
  labs(x = "Reads per samples", 
       y = "Count");Dep_raw
```
