---
title: "Summaries"
output: html_document
---

Now that we've cleaned up our taxonomy, let's take a look at the read distribution across the samples and remove samples with low read depths. 

Let's load our latest Phyloseq object 
```{r}
combinedPS_clean<-readRDS("Processed_data/combinedPS_clean.rds");combinedPS_clean
```

and needed packages 
```{r, message=FALSE,echo=TRUE}
library("microbiome") 
library("microbiomeutilities")
library("cowplot")
library("ggpubr")
library("phyloseq")
```

Taxa filtering
---- 

As mentioned previously, there are plenty of packages available to analyse metabaroding data. To summarize the data, the [Microbiome](https://microbiome.github.io/tutorials/cleaning_taxonomy_table.html) package provide a simple command to get basic information about the data.  

This will provide:
- min, max, total, mean and median number of reads
- number of singletons and low occurence ASVs (OTUs)
- data sparsity (zero inflated data)

```{r}
str(microbiome::summarize_phyloseq(combinedPS_clean))
```

Since we have a more than 1600 singletons, lets remove the taxa that are not supported by at least 10 occurrences over the whole data set. This will help us control for technical errors, even if such a threshold remain subjective (see [Rollin, Rong and Massart, 2023](https://bmcbiol.biomedcentral.com/articles/10.1186/s12915-023-01708-w) for an interesting discussion on technical errors control) and [Zhou et al. 2023](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10569954/) for examples of filtering thresholds. 

```{r}
combinedPS_clean_pruned = prune_taxa(taxa_sums(combinedPS_clean) > 09, combinedPS_clean); combinedPS_clean_pruned
```

Lets have another look at the data summary: 

```{r}
str(microbiome::summarize_phyloseq(combinedPS_clean_pruned))
```

Sample filtering
---- 

Another important thing is to identify and remove samples with low read counts. We can first plot the read distribution per sample using the plot_read_distribution function of the [microbiomutilities](https://microsud.github.io/microbiomeutilities/articles/microbiomeutilities.html) package.
There are no 

```{r, message=FALSE}
combinedPS_clean_DEP_raw<-plot_read_distribution(combinedPS_clean, 
                                groups = "seq_tech", 
                                plot.type = "histogram")+
  theme_biome_utils()+
  scale_x_continuous(trans='log10',
                   limits=c(1, 500000))+
  scale_fill_manual(values=c("#111111"))+
  geom_vline(xintercept = 1000, 
             colour = "black", 
             linetype="dashed")+
  theme(legend.position="none")+
  labs(x = "Reads per samples", 
       y = "Count");combinedPS_clean_DEP_raw
```





