---
title: "Sequencing technology comparison"
output: html_document
---

```{r setup, include=FALSE} 
knitr::opts_chunk$set(warning = FALSE, message = FALSE) 
```

```{r, message=FALSE,echo=FALSE}
get_cols <- function (n){
  col <- c("#8dd3c7", "#ffffb3", "#bebada", "#fb8072", "#80b1d3",
           "#fdb462", "#b3de69", "#fccde5", "#d9d9d9", "#bc80bd",
           "#ccebc5", "#ffed6f")
  
  col2 <- c("#1f78e4", "#ffff33", "#c2a5cf", "#ff7f00", "#810f7c",
            "#a6cee3", "#006d2c", "#4d4d4d", "#8c510a", "#d73027",
            "#78c679", "#7f0000", "#41b6c4", "#e7298a", "#54278f")
  
  col3 <- c("#a6cee3", "#1f78b4", "#b2df8a", "#33a02c", "#fb9a99",
            "#e31a1c", "#fdbf6f", "#ff7f00", "#cab2d6", "#6a3d9a",
            "#ffff99", "#b15928")
  colorRampPalette(col2)(n)
}
```

```{r, message=FALSE,echo=TRUE}
library("MicrobiotaProcess")
library("ggpubr")
library("ape")
library("UpSetR")
library("vegan")

WorkingPS<-readRDS("Processed_data/WorkingPS.rds")
```

To compare our sequencing methods, we first need to use the taxonomy; we can't make the comparison at the ASV level since they are not constant across data sets. 

The get_taxadf function in the MicrobiotaProcess package is used to aggregate data with the same taxonomy. The tax_glom function in the phyloseq package also performs this task. We will also have to rebuild the tree and our phyloseq object. 

```{r}
WorkingPS_glom <- get_taxadf(WorkingPS, 
                     taxlevel=7,
                     type = "species")



WorkingPS_glom_tree = rtree(ntaxa(WorkingPS_glom), 
                    rooted=TRUE, 
                    tip.label=taxa_names(WorkingPS_glom))


WorkingPS_glom2 <- phyloseq(phyloseq::tax_table(WorkingPS_glom),
                            phyloseq::otu_table(WorkingPS_glom),
                            WorkingPS_glom_tree,
                            phyloseq::sample_data(WorkingPS_glom))

```

PCA 
----

We can then use our agglomerated taxonomy to compare our 3 sequencing methods using principal component analysis (PCA). 

Once again, we'll use the MicrobiotaProcess package to obtain and plot the results of our PCA. 

* Note here that we have used the robust centered log ratio (rclr), a transformation method commonly used in compositional data analysis. 

```{r}
WorkingPS_glom2_PCA <- get_pca(obj=WorkingPS_glom2, 
                               method=("rclr"))


pcaplot1 <-ggordpoint(obj=WorkingPS_glom2_PCA, pc=c(1, 2), biplot=FALSE, speciesannot=FALSE,
                      factorNames=c("seq_tech"), ellipse=TRUE)+
  scale_fill_manual(values=get_cols(6))+
    theme(plot.title = element_blank())+
  theme(legend.position="Null")

pcaplot2 <- ggordpoint(obj=WorkingPS_glom2_PCA, pc=c(1, 3), biplot=FALSE, speciesannot=FALSE,
                       factorNames=c("seq_tech"), ellipse=TRUE) +
    theme(plot.title = element_blank())+
  scale_fill_manual(values=get_cols(6))

pcaplot1 | pcaplot2

```


We can see from the graph that the data are fairly clustered by technology, that nanopore and illumina data are fairly close together, and that IonTorrent data are more scattered. 

Let's try a principal coordinate analysis (PCoA).

PCOA 
---- 

Similarly, let's use the get_pcoa function (MicrobiotaProcess) to obtain and plot our PCoA results.

* Note here that we have used the wieghted unifrac distance method, which account for phylogenetic distance and abundance.  

```{r}
pcoa_results <- get_pcoa(obj=WorkingPS_glom2, 
                      distmethod="wunifrac", 
                      method="hellinger")

pcoaplot1 <- ggordpoint(obj=pcoa_results,
                          pc=c(1, 2),
                          biplot=FALSE,
                          factorNames=c("seq_tech"), 
                          ellipse=TRUE,
                          poinsize = 1.8,
                          stroke = 0.2)+
  theme(legend.box = "horizontal", legend.position = "Null")+
  scale_fill_manual(values=get_cols(6))+
  theme(legend.text=element_text(size=8))+
  theme(legend.title = element_blank())+ 
  theme(plot.title = element_blank())+
  guides(fill = guide_legend(keywidth = 0.6, 
                             keyheight = 0.7,
                             ncol = 1))

pcoaplot2 <- ggordpoint(obj=pcoa_results,
                          pc=c(1, 3),
                          biplot=FALSE,
                          factorNames=c("seq_tech"), 
                          ellipse=TRUE,
                          poinsize = 1.8,
                          stroke = 0.2)+
  theme(legend.box = "horizontal", legend.position = "right")+
  scale_fill_manual(values=get_cols(6))+
  theme(legend.text=element_text(size=8))+
  theme(legend.title = element_blank())+ 
  theme(plot.title = element_blank())+
  theme(legend.direction = "vertical", legend.box = "vertical")+
  guides(fill = guide_legend(keywidth = 0.6, 
                             keyheight = 0.7,
                             ncol = 1))
pcoaplot1 | pcoaplot2

```

Once again, we can see that the data are grouped by technology, and that Illumina's data seem closer to Nanopore's than IonTorrent's.  

Upset Plot 
----

Like Venn diagrams, UpSet graphs are a data visualization method for quantifying elements shared between and unique to given elements of comparison.

We use the get_upset function from the MicrobiotaProcess package to create our figure. 

```{r}
upsetda <- get_upset(obj=WorkingPS_glom2, factorNames="seq_tech")

upset(upsetda, sets=unique(as.vector(sample_data(WorkingPS_glom2)$seq_tech)), 
      sets.bar.color = "#56B4E9",
      order.by = "freq", 
      empty.intersections = "on")

```

We can see from this graph that 22 taxa are shared by the 3 technologies, while 35 taxa are unique to IonTorrent and 19 taxa to Nanopore.  

Permanova
---- 





```{r}
dist_mat <- get_dist(WorkingPS_glom2, distmethod ="unifrac", method="hellinger")
sampleda <- data.frame(sample_data(WorkingPS_glom2), check.names=FALSE)
sampleda <- sampleda[match(colnames(as.matrix(dist_mat)),rownames(sampleda)),,drop=FALSE]
sampleda$seq_tech <- factor(sampleda$seq_tech)
set.seed(1024)
adonis_res <- adonis(dist_mat ~ seq_tech*Week, data=sampleda, permutation=9999)
data.frame(adonis_res$aov.tab)

```
































```{r}
##### Adonis for diseased plantations vs forest  ####
diseased_forest<- subset_samples(pooledN_soil, Status !="Healthy_plantation")

distmeDF <- get_dist(diseased_forest, distmethod ="wunifrac", method="hellinger")
sampledaDF <- data.frame(sample_data(diseased_forest), check.names=FALSE)
sampledaDF <- sampledaDF[match(colnames(as.matrix(distmeDF)),rownames(sampledaDF)),,drop=FALSE]
sampledaDF$Status <- factor(sampledaDF$Status)

set.seed(1024)
adonis_DF <- adonis(distmeDF ~ Status, 
                               data=sampledaDF, 
                               permutation=9999)

pairwise_p<-numeric()
pairwise_p["DF"] <- adonis_DF[["aov.tab"]][["Pr(>F)"]][1]
```



```{r}
##### Adonis for healthy plantations vs forest  #####
healthy_forest<- subset_samples(pooledN_soil, Status !="Diseased_plantation")

distmeHF <- get_dist(healthy_forest, distmethod ="wunifrac", method="hellinger")
sampledaHF <- data.frame(sample_data(healthy_forest), check.names=FALSE)
sampledaHF <- sampledaHF[match(colnames(as.matrix(distmeHF)),rownames(sampledaHF)),,drop=FALSE]
sampledaHF$Status <- factor(sampledaHF$Status)

set.seed(1024)
adonis_HF <- adonis(distmeHF ~ Status, 
                               data=sampledaHF, 
                               permutation=9999)
adonis_HF

pairwise_p["HF"] <- adonis_HF[["aov.tab"]][["Pr(>F)"]][1]
```


```{r}
##### Adonis for diseased vs healthy plantations ####
diseased_healthy<- subset_samples(pooledN_soil, Status !="Healthy_forest")

distmeDH <- get_dist(diseased_healthy, distmethod ="wunifrac", method="hellinger")
sampledaDH <- data.frame(sample_data(diseased_healthy), check.names=FALSE)
sampledaDH <- sampledaDH[match(colnames(as.matrix(distmeDH)),rownames(sampledaDH)),,drop=FALSE]
sampledaDH$Status <- factor(sampledaDH$Status)

set.seed(1024)
adonis_DH <- adonis2(distmeDH ~ Status*Specie*Region, 
                    data=sampledaDH, 
                    permutation=9999)

adonis_DH

pairwise_p["D_H"] <- adonis_DH[["aov.tab"]][["Pr(>F)"]][1]
```



```{r}
p.adjust(pairwise_p, method="hochberg")
```







